kong/pdk/log.lua

```diff
@@ -549,25 +548,24 @@ local function get_default_serialize_values()
 end
 
 ---
--- Sets a value to be used on the `serialize` custom table
+-- Sets a value to be used on the `serialize` custom table.
 --
 -- Logging plugins use the output of `kong.log.serialize()` as a base for their logs.
+-- This function let you customize the log output.
 --
--- This function allows customizing such output.
---
--- It can be used to replace existing values on the output.
--- It can be used to delete existing values by passing `nil`.
+-- It can be used to replace existing values in the output, or to delete
+-- existing values by passing `nil`.
 --
--- Note: the type checking of the `value` parameter can take some time so
+-- **Note:** The type-checking of the `value` parameter can take some time, so
 -- it is deferred to the `serialize()` call, which happens in the log
 -- phase in most real-usage cases.
 --
 -- @function kong.log.set_serialize_value
 -- @phases certificate, rewrite, access, header_filter, response, body_filter, log
--- @tparam string key the name of the field.
--- @tparam number|string|boolean|table value value to be set. When a table is used, its keys must be numbers, strings, booleans, and its values can be numbers, strings or other tables like itself, recursively.
--- @tparam table options can contain two entries: options.mode can be `set` (the default, always sets), `add` (only add if entry does not already exist) and `replace` (only change value if it already exists).
--- @treturn table the request information table
+-- @tparam string key The name of the field.
+-- @tparam number\|string\|boolean\|table value Value to be set. When a table is used, its keys must be numbers, strings, or booleans, and its values can be numbers, strings, or other tables like itself, recursively.
```

Comment:
Is there a reason these `|` chars are being escaped like this? I ask because the syntax looks correct to me as-is, where `@tparam a|b|c` implies a union type ("the function accepts type `a`, type `b`, or type `c` for this parameter").

Reply:
These comments get converted to markdown for the doc output. If they aren't escaped or enclosed in backticks, any time that `|` gets used more than once in a line generates a table row. For example, see the value entry here: https://docs.konghq.com/gateway/2.7.x/pdk/kong.response/#kongresponseadd_headername-value

Reply:
Okay gotcha. Hmm.

Besides being a source from which the html/markdown docs are generated, these annotations (ldoc/luadoc) also serve as an inline reference for developers when reading the code directly, so I'd personally rather not have this escaping done in the source code. It may also break some text editor/IDE/language server implementations that parse these annotations in order to provide type hints and code-completion.

Can we perform this escaping during doc generation instead? I haven't really dug into our ldoc templates all that much, but I believe it should be possible with a strategic placement of something like `string.gsub(parameter_type, "|", "\|")`

https://github.com/Kong/kong/blob/master/autodoc/pdk/ldoc/ldoc.ltp

Reply:
Definitely possible, let me give it a shot.

Reply:
Applied all of your other feedback but struggling to figure out how/where exactly to do this. Any ideas?

Reply:
Oop, didn't have time to circle back to this today, but I'll try to take a look soon (also pinging the team in case someone more ldoc-savvy feels like chiming in)

Reply:
Interesting. It seems preventing that was the idea behind this bit of template code, but it seems that is not enough:

```
>               if type ~= '' then
>                 type = "`" .. type .. "`:"
>               end
```

Let me investigate (having a link to a place where the generated code breaks the table is very helpful for that)

Reply:
I was incorrect, that piece of code was for the __return__ values, and it was working as indended.

Our __parameters__ don't have back-ticks around their types. I have added a change on this PR that should fix that: it will surround all the parameter types with back-ticks, unless they already have them.

Reply:
Just ran a quick test to generate these docs, everything looks good. Thanks for the help! 🙏